src/scripts/arabic.rs

```diff
@@ -0,0 +1,140 @@
+const NONE: u32 = u32::MAX;
```

Comment:
[NOTE] This implementation was inspired by [this library](https://github.com/eloraiby/arabtype/blob/master/arabtype.c). 🙏

Reply:
Thanks for being transparent about this! :)